CMakeLists.txt

```diff
@@ -1,355 +1,314 @@
-    PathOrderOptimizerTest
-    PathOrderMonotonicTest
-    TimeEstimateCalculatorTest
-    WallsComputationTest
-)
+        ClipperTest
+        ExtruderPlanTest
+        GCodeExportTest
+        InfillTest
+        LayerPlanTest
+        PathOrderOptimizerTest
+        PathOrderMonotonicTest
+        TimeEstimateCalculatorTest
+        WallsComputationTest
+        )
 if (ENABLE_ARCUS)
     set(engine_TEST_ARCUS
-        ArcusCommunicationTest
-        ArcusCommunicationPrivateTest
-    )
+            ArcusCommunicationTest
+            ArcusCommunicationPrivateTest
+            )
 endif ()
 set(engine_TEST_BEADING_STRATEGY
-    CenterDeviationBeadingStrategyTest
-)
+        CenterDeviationBeadingStrategyTest
+        )
 set(engine_TEST_INFILL
-)
+        )
 set(engine_TEST_INTEGRATION
-    SlicePhaseTest
-)
+        SlicePhaseTest
+        )
 set(engine_TEST_SETTINGS
-    SettingsTest
-)
+        SettingsTest
+        )
 set(engine_TEST_UTILS
-    AABBTest
-    AABB3DTest
-    ExtrusionLineTest
-    IntPointTest
-    LinearAlg2DTest
-    MinimumSpanningTreeTest
-    PolygonConnectorTest
-    PolygonTest
-    PolygonUtilsTest
-    SparseGridTest
-    StringTest
-    UnionFindTest
-)
+        AABBTest
+        AABB3DTest
+        ExtrusionLineTest
+        IntPointTest
+        LinearAlg2DTest
+        MinimumSpanningTreeTest
+        PolygonConnectorTest
+        PolygonTest
+        PolygonUtilsTest
+        SparseGridTest
+        StringTest
+        UnionFindTest
+        )
 
 # Helper classes for some tests.
 set(engine_TEST_ARCUS_HELPERS
-    tests/arcus/MockSocket.cpp
-)
+        tests/arcus/MockSocket.cpp
+        )
 set(engine_TEST_HELPERS
-    tests/ReadTestPolygons.cpp
-)
+        tests/ReadTestPolygons.cpp
+        )
 
-# Generating ProtoBuf protocol
-if (ENABLE_ARCUS)
-    protobuf_generate_cpp(engine_PB_SRCS engine_PB_HEADERS Cura.proto)
-endif ()
+add_library(_CuraEngine STATIC ${engine_SRCS} ${engine_PB_SRCS})
+use_threads(_CuraEngine)
 
-# Compiling CuraEngine itself.
-add_library(_CuraEngine STATIC ${engine_SRCS} ${engine_PB_SRCS}) #First compile all of CuraEngine as library, allowing this to be re-used for tests.
+target_include_directories(_CuraEngine
+        PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}> # Include Cura.pb.h
+        )
+target_compile_definitions(_CuraEngine
+        PUBLIC
+        $<$<BOOL:${BUILD_TESTS}>:BUILD_TESTS=1>
+        $<$<BOOL:${BUILD_TESTS}>:BUILD_TESTS>
+        $<$<BOOL:${ENABLE_ARCUS}>:ARCUS>
+        PRIVATE
+        VERSION=\"${CURA_ENGINE_VERSION}\"
+        $<$<BOOL:${WIN32}>:NOMINMAX>
+        $<$<CONFIG:Debug>:ASSERT_INSANE_OUTPUT>
+        $<$<CONFIG:Debug>:USE_CPU_TIME>
+        $<$<CONFIG:Debug>:DEBUG>
+        $<$<CONFIG:RelWithDebInfo>:ASSERT_INSANE_OUTPUT>
+        $<$<CONFIG:RelWithDebInfo>:USE_CPU_TIME>
+        $<$<CONFIG:RelWithDebInfo>:DEBUG>
```

Comment:
I think perhaps just `if(Debug OR RelWithDebInfo)` is easier syntax to read than this.

A bit of a matter of style though, so I'm fine either way.